packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -1875,7 +1886,8 @@ class _InputDecoratorState extends State<InputDecorator> with TickerProviderStat
     _floatingLabelController = AnimationController(
       duration: _kTransitionDuration,
       vsync: this,
-      value: (widget.decoration.hasFloatingPlaceholder && widget._labelShouldWithdraw) ? 1.0 : 0.0,
+      // ignore: deprecated_member_use_from_same_package
+      value: widget.decoration.floatLabelBehavior == FloatLabelBehavior.always || (widget.decoration.hasFloatingPlaceholder && widget._labelShouldWithdraw) ? 1.0 : 0.0,
```

Comment:
Although it's true that wide lines are OK, the overall goal is for the code to be easy to read (see https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#prefer-a-maximum-line-length-of-80-characters). In this case using two lines would be a little better:

```dart
value: widget.decoration.floatLabelBehavior == FloatLabelBehavior.always
  || (widget.decoration.hasFloatingPlaceholder && widget._labelShouldWithdraw) ? 1.0 : 0.0,
```

Introducing a variable would help a little as well, since the variable name helps explain what's meant:

```dart
final bool labelIsInitiallyFloating = widget.decoration.floatLabelBehavior == FloatLabelBehavior.always
 || (widget.decoration.hasFloatingPlaceholder && widget._labelShouldWithdraw);

value: labelIsInitiallyFloating: ? 1.0 : 0.0;
```